data/lib/core/tile.lua

```diff
@@ -33,19 +33,43 @@ function Tile.relocateTo(self, toPosition)
 	return true
 end
 
-function Tile.isWalkable(self)
-	local ground = self:getGround()
-	if not ground or ground:hasProperty(CONST_PROP_BLOCKSOLID) then
-		return false
-	end
+function Tile.isWalkable(self, check)
+  local ground = self:getGround()
+  if not ground or ground:hasProperty(CONST_PROP_BLOCKSOLID) then
+    return false
+  end
 
-	local items = self:getItems()
-	for i = 1, self:getItemCount() do
-		local item = items[i]
-		local itemType = item:getType()
-		if itemType:getType() ~= ITEM_TYPE_MAGICFIELD and not itemType:isMovable() and item:hasProperty(CONST_PROP_BLOCKSOLID) then
-			return false
-		end
-	end
-	return true
+  if check then
+    if check.protection and self:hasFlag(TILESTATE_PROTECTIONZONE) or
+      check.floorChange and self:hasFlag(TILESTATE_FLOORCHANGE) or
+      check.house and self:getHouse() ~= nil then
```

Comment:
I see. I copied this from getTileInfo(position) of compat.lua.
I will change this, but I suggest you to remove there also. :)